src/zenml/post_execution/pipeline.py

```diff
@@ -134,11 +125,78 @@ def get_pipeline(
         return None
 
 
+class PipelineClassView:
+    """Post-execution class for a pipeline name/class."""
+
+    def __init__(self, name: str):
+        """Initializes a post-execution class for a pipeline name/class.
+
+        Args:
+            name: The name of the pipeline.
+        """
+        self.name = name
+
+    @property
+    def versions(self) -> List["PipelineView"]:
+        """Returns all versions/instances of this pipeline name/class.
+
+        Returns:
+            A list of all versions of this pipeline.
+        """
+        client = Client()
+        pipelines = client.list_pipelines(
+            workspace_id=client.active_workspace.id,
+            name=self.name,
+            sort_by="desc:created",
+        )
+        return [PipelineView(model) for model in pipelines.items]
+
+    @property
+    def num_runs(self) -> int:
+        """Returns the number of runs of this pipeline name/class.
+
+        This is the sum of all runs of all versions of this pipeline.
+
+        Returns:
+            The number of runs of this pipeline name/class.
+        """
+        return sum([version.num_runs for version in self.versions])
```

Comment:
```suggestion
        return sum(version.num_runs for version in self.versions)
```

Save a bit of memory with a generator expression instead of a list comprehension?